Code/Framework/AzCore/AzCore/Name/Name.cpp

```diff
@@ -77,19 +78,28 @@ namespace AZ
         return m_data == nullptr ? 0 : m_data->GetHash();
     }
 
-    Name::Name()
-        : m_view("")
-    {
-    }
+    Name::Name() = default;
 
     Name::Name(AZStd::string_view name)
     {
         SetName(name);
     }
+    Name::Name(AZStd::string_view name, NameDictionary& nameDictionary)
+    {
+        SetName(name, nameDictionary);
+    }
 
     Name::Name(Hash hash)
     {
-        *this = NameDictionary::Instance().FindName(hash);
+        auto nameDictionary = AZ::Interface<NameDictionary>::Get();
+        AZ_Assert(nameDictionary != nullptr, "hash value %u cannot be looked up before the global before the NameDictionary is ready.\n"
+            "If a explicit name dictionary is available, it can be passed to the Name(Hash, NameDictionary&) overload instead.", hash);
+        *this = nameDictionary->FindName(hash);
```

Comment:
```suggestion
        AZ_Assert(nameDictionary != nullptr, "hash value %u cannot be looked up before the global NameDictionary is ready.\n"
            "If an explicit name dictionary is available, it can be passed to the Name(Hash, NameDictionary&) overload instead.", hash);
        *this = nameDictionary->FindName(hash);
```